generators/Reporting.fs

```diff
@@ -0,0 +1,92 @@
+module Generators.Reporting
+
+open Serilog
+open CanonicalData
+open Exercise
+open Options
+       
+let private isOutdated (exercise: GeneratorExercise) options (parseCanonicalData':string -> CanonicalData) =
+    let cData = parseCanonicalData' exercise.Name
+    match cData.Version,exercise.ReadVersion() with
+    | canonVersion,exerciseVersion when canonVersion.Equals exerciseVersion -> false
+    | _,_ -> true
+    
+let private filterByStatus options (parseCanonicalData':string -> CanonicalData) (exercise: Exercise) =
+    match options.Status, exercise with
+    | None, _ -> true
+    | Some Status.Implemented,   Exercise.Generator _     -> true
+    | Some Status.Unimplemented, Exercise.Unimplemented _ -> true
+    | Some Status.MissingData,   Exercise.MissingData _   -> true
+    | Some Status.Deprecated,    Exercise.Deprecated _    -> true
+    | Some Status.Custom,        Exercise.Custom _        -> true
+    | Some Status.Outdated,      Exercise.Generator exercise when isOutdated exercise options parseCanonicalData' -> true
+    | Some Status.All,           _                        -> true
+    | _ -> false
+    
+type SummaryTypes =
+    | Custom of string
+    | Unimplemented of string
+    | MissingData of string
+    | Deprecated of string
+    | UpToDate of string
+    | Outdated of string * string * string
+    
+let nameFromSummaryType = function
+    | Custom n -> n
+    | Unimplemented  n -> n
+    | MissingData n -> n
+    | Deprecated n -> n
+    | UpToDate n -> n
+    | Outdated (n,_,_) -> n
```

Comment:
Could you add a spaces after the commas?

Reply:
The community really needs some better lint tooling and autoformatting for this kind of stuff 😐

Reply:
Absolutely agree! Maybe we should take a look at integrating [fantomas](https://github.com/fsprojects/fantomas) sometime.